server/index.js

```diff
@@ -55,6 +55,103 @@ app.get('/api/products/:productId', function (req, res, next) {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', function (req, res, next) {
+  if (!req.session.cartId) {
+    res.status(200).json([]);
+  } else {
+    const sql = `
+    select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using ("productId")
+      where "c"."cartId" = $1
+    `;
+    const params = [req.session.cartId];
+    db.query(sql, params)
+      .then(result => res.status(200).json(result.rows))
+      .catch(err => next(err));
+  }
+});
+
+app.post('/api/cart', function (req, res, next) {
+  const productId = Number(req.body.productId);
+  if (isNaN(productId) || !Number.isInteger(productId) || productId <= 0) {
+    return res.status(400).json({ error: 'productId must be a positive integer' });
+  }
+  const sql = `
+  select "price"
+  from "products"
+  where "productId" = $1
+  `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0]) {
+        next(new ClientError(`cannot find product with 'productId' ${productId}`, 400));
```

Comment:
Instead of calling `next()`, we want to abort the promise chain. Calling `next()` does not cause this JavaScript code to stop executing. Whereas `throw`ing will shortcut us to the `.catch()`.

```js
throw new ClientError(message, status)
```